sys/compat/linprocfs/linprocfs.c

```diff
@@ -2092,6 +2095,158 @@ linprocfs_domax_map_cnt(PFS_FILL_ARGS)
 	return (0);
 }
 
+/*
+ * Filler function for proc/sysvipc/msg
+ */
+static int
+linprocfs_dosysvipc_msg(PFS_FILL_ARGS)
+{
+	struct msqid_kernel *msqids;
+	u_long id, msgmni;
+	size_t sz;
+	int error;
+
+	sbuf_printf(sb,
+	    "%10s %10s %4s  %10s %10s %5s %5s %5s %5s %5s %5s %10s %10s %10s\n",
+	    "key", "msqid", "perms", "cbytes", "qnum", "lspid", "lrpid",
+	    "uid", "gid", "cuid", "cgid", "stime", "rtime", "ctime");
+
+	sz = sizeof(struct msqid_kernel) * msginfo.msgmni;
```

Comment:
msginfo is not available if sysvmsg module is not compiled into the kernel static.  You need to add dependency on sysvmsg module and check that linprocfs is still loadable even if the sysvmsg module is not statically compiled.

Reply:
> msginfo is not available if sysvmsg module is not compiled into the kernel static. You need to add dependency on sysvmsg module and check that linprocfs is still loadable even if the sysvmsg module is not statically compiled.

There are 3 lines at EOF:

```
MODULE_DEPEND(linprocfs, sysvmsg, 1, 1, 1);
MODULE_DEPEND(linprocfs, sysvsem, 1, 1, 1);
MODULE_DEPEND(linprocfs, sysvshm, 1, 1, 1);
```

Reply:
ok, sorry